server-sdk/src/main/java/com/amazonaws/ml/mms/servingsdk/http/Request.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.amazonaws.ml.mms.servingsdk.http;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This defines the request object given to the custom endpoint
+ */
+public interface Request {
+    /**
+     * Get all header names in the request object
+     * @return
+     */
+    List<String> getHeaderNames();
+
+    /**
+     * Get the URI of the request
+     * @return URI
+     */
+    String getRequestURI();
+
+    /**
+     * Get the path of the endpoint
+     * @return
+     */
+    String getRequestPath();
+
+    /**
+     * Get all query parameters coming in for this endpoint
+     * @return
+     */
+    Map<String, List<String>> getParameterMap();
+
+    /**
+     * Get a query parameter
+     * @param k - Parameter name
+     * @return - value of the parameter
+     */
+    List<String> getParameter(String k);
+
+    /**
+     * Get the content-type of the incoming request object
+     * @return
+     */
+    String getContentType();
+
+    /**
+     * Get the body content stream of the incoming request
+     * @return
+     * @throws IOException
+     */
+    InputStream getInputStream() throws IOException;
```

Comment:
Did you removed getMethod()?

Reply:
Yes. I realized that we have individual calls like "doGet", "doPost". So, if the "doGet" is being called, then it implicitly means its a get method. We can add the "getMethod", once we introduce "doService". I don't see a reason to have it until then.